src/components/candevice/candevice.cpp

```diff
@@ -100,6 +128,43 @@ QJsonObject CanDevice::getConfig() const
     return {};
 }
 
+void CanDevice::setConfig(const QObject& qobject)
+{
+    Q_D(CanDevice);
+
+    for (const auto& p: getSupportedProperties())
+    {
+        QVariant v = qobject.property(p.first.toStdString().c_str());
+        if (!v.isValid() || v.type() != p.second.first)
```

Comment:
I would explicitly express the dependent content under `if`, rather that use `continue`, i.e.:

```
auto v = ...
if (...) { d->... }
```